### ISSUE TITLE
onPaymentDetected error

### ISSUE THREAD

**Issue by Sm00g15** [coinbase/react-coinbase-commerce#13]
Whenever a payment is detected, I get a TypeError: 

`checkout.b56640fe.js:1 TypeError: Cannot read property 'payment_detected' of undefined`

My component looks like so: 

```
 <CoinbaseCommerceButton
                        checkoutId="12a69b79-11bc-420b-****-06a13edddf6f"
                        styled={true}
                        onModalClosed={() => this.props.modalBody("")}
                        onPaymentDetected={res => {
                          console.log(res);
                        }} 
/>
```

Even with out console logging the res I'm hitting that same error. Can you show me how to handle responses with onPaymentDetected so that I can use that data to manipulate the DOM please?

looking to do something like this: 
onPaymentDetected={res => {if(res.data === "success"} {*close modal*, *make it so next time the Pay Now button is clicked a new charge is created so the user can make many purchases in relatively rapid succession*}}

**Comment by tysonbattistella** [comment]
@Sm00g15 Thanks for pointing this out! This was an internal bug, and should be fixed now. As for the second issue, I'm unable to replicate it locally. Do you have any reproduction steps I could try? Closing this for now, feel free to create a new issue for the second one if you're still encountering it.

**Comment by Sm00g15** [comment]
Hey Tyson,

I resubmitted a new issue describing my second issue more succinctly. The TypeError in my first issue has gone away, so thanks for that! I wasn’t getting metadata back though with the last payment I made, and i’ll be further testing that to make sure it goes through, catch is I can’t test because now when I open up the modal it only shows the completion screen! As soon as I get past that and can make a new charge though I’ll let you know if the customMetadata field still works. Thanks again!
